src/huggingface_hub/hf_api.py

```diff
@@ -622,6 +622,53 @@ def clean(s: str):
         self["author"] = author_dict
 
 
+@dataclass
+class RefInfo:
+    """
+    Contains information about a git reference for a repo on the Hub.
+
+    Args:
+        name (`str`):
+            Name of the reference (e.g. tag name or branch name).
+        ref (`str`):
+            Full git ref on the Hub (e.g. `"refs/heads/main"` or `"refs/tags/v1.0"`).
+        target_commit (`str`):
+            OID of the target commit for the ref (e.g. `"e7da7f221d5bf496a48136c0cd264e630fe9fcc8"`)
+    """
+
+    name: str
+    ref: str
+    target_commit: str
+
+    @classmethod
+    def from_dict(cls, data: Dict) -> "RefInfo":
```

Comment:
you're not just overriding the init from dataclass?

Reply:
(dont remember the best practice in Python dataclasses)

Reply:
Yep why not. Best practice depends on your use case I guess.

Here the `GitRefs` will never be initialized by someone else than use from server-side data so it makes sense to put it in the init (I made the change). It's not good practice when you force the users to have your data format as input instead of letting the default dataclass constructor.